### PR TITLE
[API-525] [API-526] Add documentation and code samples for the compact serialization

### PR DESCRIPTION
**Pull request by mdumandag**

### CODE REVIEW

**Comment by mdumandag** [comment]
Note to myself, we should document `compact_serializers` in the client constructor

File: docs/serialization.rst [hunk 1]
```diff
@@ -46,35 +46,256 @@ When Hazelcast Python client serializes an object:
...
+        def get_type_name(self) -> str:
+            return "employee"
+
+The last step is to register the serializer in the client configuration.
+
+.. code:: python
+
+    client = HazelcastClient(
+        compact_serializers={
+            Employee: EmployeeSerializer(),
```
Comment (yuce) [inline docs/serialization.rst:170]: I thought we decided to change this.

Comment (mdumandag) [reply docs/serialization.rst:170]: I thought we said we would stick to this in our discussion after seeing Sancar's message

Comment (yuce) [reply docs/serialization.rst:170]: I'll resolve this so we can merge the PR. I'll send a separate PR to sync this with the latest decision.

**Review by srknzl — APPROVED** [review]

**Pull request merged**